### DISCUSSION TITLE
How to run diktat only for new changes?

### DISCUSSION THREAD

**Discussion by akshatagarwl** [cqfn/diKTat#1275] [Q&A] [answered]
Introducing a linter to new project becomes a huge task especially fixing all the existing issues first. Is there a way to run diktat only for new changes and add that check in the CI so that any new code that has been added to a codebase doesn't violate the rules set by diktat?

**Comment by orchestr7** [comment, marked as answer]
You can use baseline for this purpose: https://github.com/analysis-dev/diktat#running-against-the-baseline

```

<?xml version="1.0" encoding="utf-8"?>
<baseline version="1.0">
    <file name="src/test/resources/TestBaselineFile.kt">
        <error line="1" column="24" source="no-empty-class-body" />
        <error line="2" column="1" source="no-blank-line-before-rbrace" />
    </file>
    <file name="src/test/resources/TestBaselineExtraErrorFile.kt">
        <error line="1" column="34" source="no-empty-class-body" />
    </file>
</baseline>

```
